### PR TITLE
Compliance tests jsonfix

### PR DESCRIPTION
**Pull request by yisraelU**
adds dependency on circe for the Compliance tests module
detects when content type is json and the comparison is between 2 strings parse them first using a json parser to remove insignificant characters

### CODE REVIEW

**Comment by kubukoz** [comment]
Q: how about, instead of a json AST, we parse the json into a Document and compare these instead? That'd avoid the need to depend on Circe

**Comment by Baccata** [comment]
> Q: how about, instead of a json AST, we parse the json into a Document and compare these instead? That'd avoid the need t!o depend on Circe

 We discussed it internally and this basically implies that we'd be testing smithy4s behaviour (json decoding) with smithy4s constructs, which ain't a good idea.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -36,6 +36,12 @@ object Dependencies {
       Def.setting("org.typelevel" %%% "cats-core" % "2.8.0")
   }
 
+  object Circe {
+    val circeVersion = "0.14.1"
+    val parser = "io.circe" %% "circe-parser" % circeVersion
```
Comment (Baccata) [inline project/Dependencies.scala:41]: http://sca.la/percent

File: project/Dependencies.scala [hunk 2]
```diff
@@ -37,9 +37,9 @@ object Dependencies {
   }
 
   object Circe {
-    val circeVersion = "0.14.1"
-    val parser = "io.circe" %% "circe-parser" % circeVersion
-    val generic = "io.circe" %% "circe-generic" % circeVersion
+    val circeVersion = "0.14.3"
+    val parser =  Def.setting("io.circe" %%% "circe-parser" % circeVersion)
```
Comment (daddykotex) [inline project/Dependencies.scala:41]: ahh, is the `Def.setting` required when using `%%%` ?

**Review by daddykotex — APPROVED** [review]
👍

**Pull request merged**